File: examples/tezos.py
from util import *
from irmin import *
import sys

if len(sys.argv) < 2:
    print("usage: python3 tezos.py /path/to/tezos/context [chain_id]")
    sys.exit(0)

root = sys.argv[1]

# Configure tezos store
config = Config.tezos(root=root)
config["readonly"] = True

# Initialize the repo
repo = Repo(config)

if len(sys.argv) > 2:
    chain_id = sys.argv[2]
else:
    chain_id = repo.branches[0]

# Open the `master` branch
store = Store(repo, branch=chain_id)


def list_path(store, path):
    '''
    Prints all content paths
    '''
    for k in store.list(path):
        p = path.append(k)
        if p in store:
            print(p)
        else:
            list_path(store, p)


# Print contract paths
list_path(store, Path(store.repo, ["data", "contracts"]))

File: rust/examples/tezos.rs
use irmin::*;

fn list_path<T: Contents>(store: &Store<T>, path: Path) -> Result<(), Error> {
    for k in store.list(&path)? {
        let p = path.append_path(&k)?;

        // If the store has contents at `p` then print the path
        if store.mem(&p) {
            println!("{}", path.to_string()?);
        } else {
            list_path(store, p)?;
        }
    }

    Ok(())
}

fn main() -> Result<(), Error> {
    let args: Vec<_> = std::env::args().collect();

    if args.len() < 2 {
        println!("usage {} /path/to/tezos/context", &args[0]);
        return Ok(());
    }

    // Configure an in-memory store with `Json` contents
    let mut config = Config::<IrminString>::tezos()?;
    config.set_root(&args[1])?;

    // Initialize the repo
    let repo = Repo::new(config)?;

    // Open the main branch
    let store = Store::of_branch(&repo, "master")?;

    // List content paths
    let path = Path::new(&repo, &["data", "contracts"])?;
    list_path(&store, path)?;

    Ok(())
}
